pagefaultgames/pokerogue
# [BUG] Fixes bug with Metronome freezing the game

## What are the changes?

#2790
Metronome calling a ChargeAttr move (e.g. Beak Blast) will freeze the game.

## Why am I doing these changes?



Fixes a bug that freezes the game.

## What did change?


I implemented a check to see if the user has the move in their moveset, if not create a dummy PokemonMove for the case of Metronome and pass it into pushMovePhase.

### Screenshots/Videos


Before fix (game freezes from getMove)

https://github.com/pagefaultgames/pokerogue/assets/62030095/05761910-0c3d-4c66-8d1a-0ac40c5c226e

After fix 

https://github.com/pagefaultgames/pokerogue/assets/62030095/34db36ab-a938-4110-9664-d9e39ca102b0

## How to test the changes?




Give your pokemon Metronome and manipulate the moveId variable in RandomMoveAttr to always use Beak Blast 
``` const moveId = moveIds[585];```
Use Metronome with Beak Blast, see that game continues properly and Beak Blast goes off.

## Checklist
